plugins/radar-android-polar/src/main/java/org/radarbase/passive/polar/PolarManager.kt

```diff
@@ -187,6 +186,33 @@ class PolarManager(
         } catch (a: PolarInvalidArgument) {
             a.printStackTrace()
         }
+    }
+
+    override fun onClose() {
+        super.onClose()
+        if (autoConnectDisposable != null && !autoConnectDisposable!!.isDisposed) {
+            autoConnectDisposable?.dispose()
+        }
+        if (hrDisposable != null && !hrDisposable!!.isDisposed) {
+            hrDisposable?.dispose()
+        }
+        if (ecgDisposable != null && !ecgDisposable!!.isDisposed) {
+            ecgDisposable?.dispose()
+        }
+        if (accDisposable != null && !accDisposable!!.isDisposed) {
+            accDisposable?.dispose()
+        }
+        if (ppiDisposable != null && !ppiDisposable!!.isDisposed) {
+            ppiDisposable?.dispose()
+        }
+        if (ppgDisposable != null && !ppgDisposable!!.isDisposed) {
+            ppgDisposable?.dispose()
+        }
+
+        disconnectToPolarSDK(deviceId)
+        mHandler.stop {
+            wakeLock?.release()
```

Comment:
Is it necessary to use a wake lock here, considering that this specific functionality is already provided by services?